simulator/.clang-format

```diff
@@ -0,0 +1,30 @@
+---
+BasedOnStyle: Google
+IndentWidth: 4
+ColumnLimit: 80
+---
+Language: Cpp
+
+AllowShortBlocksOnASingleLine: true
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortFunctionsOnASingleLine: Empty
+
+BreakBeforeBraces: Custom
+BraceWrapping:   
+  AfterClass:            false
+  AfterControlStatement: false
+  AfterEnum:             false
+  AfterFunction:         false
+  AfterNamespace:        false
+  AfterObjCDeclaration:  false
+  AfterStruct:           false
+  AfterUnion:            false
+  BeforeCatch:           false
+  BeforeElse:            true
+  IndentBraces:          false
+
+Cpp11BracedListStyle: false
+
+MaxEmptyLinesToKeep: 2
+---
+
```

Comment:
Basically, yes, there is the libavr32 submodule which is used by other modules. 

I'm just being cautious, if everyone gets on with using an editorconfig and clang-format then we'll do libavr32 as well, and then move these files to the root.